src/internet_identity/src/vc_mvp.rs

```diff
@@ -0,0 +1,247 @@
+    pub(crate) relying_party: FrontendHostname,
+    pub(crate) issuer: FrontendHostname,
+}
+
+pub async fn prepare_id_alias(
+    identity_number: IdentityNumber,
+    dapps: InvolvedDapps,
+) -> PreparedIdAlias {
+    state::ensure_salt_set().await;
+    check_frontend_length(&dapps.relying_party);
+    check_frontend_length(&dapps.issuer);
+
+    let id_alias_principal = get_id_alias_principal(identity_number, &dapps);
+    let seed = calculate_id_alias_seed(identity_number, &dapps);
+    let canister_id = ic_cdk::id();
+    let canister_sig_pk = CanisterSigPublicKey::new(canister_id, seed.to_vec());
+
+    let rp_tuple = AliasTuple {
+        id_alias: id_alias_principal,
+        id_dapp: delegation::get_principal(identity_number, dapps.relying_party.clone()),
+    };
+    let issuer_tuple = AliasTuple {
+        id_alias: id_alias_principal,
+        id_dapp: delegation::get_principal(identity_number, dapps.issuer.clone()),
+    };
+
+    let (rp_id_alias_jwt, rp_msg_hash) = id_alias_jwt_and_msg_hash(&rp_tuple, &canister_sig_pk);
+    let (issuer_id_alias_jwt, issuer_msg_hash) =
+        id_alias_jwt_and_msg_hash(&issuer_tuple, &canister_sig_pk);
+
+    state::signature_map_mut(|sigs| {
+        add_signature(sigs, rp_msg_hash, seed);
+        add_signature(sigs, issuer_msg_hash, seed);
+    });
+    update_root_hash();
+    PreparedIdAlias {
+        canister_sig_pk_der: ByteBuf::from(canister_sig_pk.to_der()),
+        rp_id_alias_jwt,
+        issuer_id_alias_jwt,
+    }
+}
+
+fn id_alias_jwt_and_msg_hash(
+    alias_tuple: &AliasTuple,
+    canister_sig_pk: &CanisterSigPublicKey,
+) -> (String, Hash) {
+    let credential_jwt = prepare_id_alias_jwt(alias_tuple);
+    let signing_input =
+        vc_signing_input(&credential_jwt, canister_sig_pk).expect("failed getting signing_input");
+    let msg_hash = vc_signing_input_hash(&signing_input);
+    (credential_jwt, msg_hash)
+}
+
+pub fn get_id_alias(
+    identity_number: IdentityNumber,
+    dapps: InvolvedDapps,
+    rp_id_alias_jwt: &str,
+    issuer_id_alias_jwt: &str,
+) -> GetIdAliasResponse {
+    check_frontend_length(&dapps.relying_party);
+    check_frontend_length(&dapps.issuer);
+
+    state::assets_and_signatures(|cert_assets, sigs| {
+        let id_alias_principal = get_id_alias_principal(identity_number, &dapps);
+        let seed = calculate_id_alias_seed(identity_number, &dapps);
+        let id_rp = delegation::get_principal(identity_number, dapps.relying_party.clone());
+        let id_issuer = delegation::get_principal(identity_number, dapps.issuer.clone());
+        let canister_id = ic_cdk::id();
+        let canister_sig_pk = CanisterSigPublicKey::new(canister_id, seed.to_vec());
+
+        let signing_input = vc_signing_input(rp_id_alias_jwt, &canister_sig_pk)
+            .expect("failed getting signing_input");
+        let msg_hash = vc_signing_input_hash(&signing_input);
+        let maybe_sig = get_signature(cert_assets, sigs, seed, msg_hash);
```

Comment:
This can be simplified to:
```
let Some(rp_sig) = get_signature(cert_assets, sigs, seed, msg_hash) else {
    return GetIdAliasResponse::NoSuchCredentials("rp_sig not found".to_string());
};
```